gnark-prover/test-data/inclusion.csv

```diff
@@ -1,167 +1,40 @@
-0;{"root": ["0x3e7","0xadcce8d93aed9fa23efe7663b02b794a5e9386973892b23d462210512838011","0xadcce8d93aed9fa23efe7663b02b794a5e9386973892b23d462210512838011","0xadcce8d93aed9fa23efe7663b02b794a5e9386973892b23d462210512838011"],"inPathIndices": [999,4,4,4],"inPathElements": [["0x3e7","0x2098f5fb9e239eab3ceac3f27b81e481dc3124d55ffed523a839ee8446b64864","0x1069673dcdb12263df301a6ff584a7ec261a44cb9dc68df067a4774460b1f1e1","0x18f43331537ee2af2e3d758d50f72106467c6eea50371dd528d57eb2b856d238","0x7f9d837cb17b0d36320ffe93ba52345f1b728571a568265caac97559dbc952a","0x2b94cf5e8746b3f5c9631f4c5df32907a699c58c94b2ad4d7b5cec1639183f55","0x2dee93c5a666459646ea7d22cca9e1bcfed71e6951b953611d11dda32ea09d78","0x78295e5a22b84e982cf601eb639597b8b0515a88cb5ac7fa8a4aabe3c87349d"],["0x3e7","0x2098f5fb9e239eab3ceac3f27b81e481dc3124d55ffed523a839ee8446b64864","0x1069673dcdb12263df301a6ff584a7ec261a44cb9dc68df067a4774460b1f1e1","0x18f43331537ee2af2e3d758d50f72106467c6eea50371dd528d57eb2b856d238","0x7f9d837cb17b0d36320ffe93ba52345f1b728571a568265caac97559dbc952a","0x2b94cf5e8746b3f5c9631f4c5df32907a699c58c94b2ad4d7b5cec1639183f55","0x2dee93c5a666459646ea7d22cca9e1bcfed71e6951b953611d11dda32ea09d78","0x78295e5a22b84e982cf601eb639597b8b0515a88cb5ac7fa8a4aabe3c87349d"],["0x3e7","0x2098f5fb9e239eab3ceac3f27b81e481dc3124d55ffed523a839ee8446b64864","0x1069673dcdb12263df301a6ff584a7ec261a44cb9dc68df067a4774460b1f1e1","0x18f43331537ee2af2e3d758d50f72106467c6eea50371dd528d57eb2b856d238","0x7f9d837cb17b0d36320ffe93ba52345f1b728571a568265caac97559dbc952a","0x2b94cf5e8746b3f5c9631f4c5df32907a699c58c94b2ad4d7b5cec1639183f55","0x2dee93c5a666459646ea7d22cca9e1bcfed71e6951b953611d11dda32ea09d78","0x78295e5a22b84e982cf601eb639597b8b0515a88cb5ac7fa8a4aabe3c87349d"],["0x3e7","0x2098f5fb9e239eab3ceac3f27b81e481dc3124d55ffed523a839ee8446b64864","0x1069673dcdb12263df301a6ff584a7ec261a44cb9dc68df067a4774460b1f1e1","0x18f43331537ee2af2e3d758d50f72106467c6eea50371dd528d57eb2b856d238","0x7f9d837cb17b0d36320ffe93ba52345f1b728571a568265caac97559dbc952a","0x2b94cf5e8746b3f5c9631f4c5df32907a699c58c94b2ad4d7b5cec1639183f55","0x2dee93c5a666459646ea7d22cca9e1bcfed71e6951b953611d11dda32ea09d78","0x78295e5a22b84e982cf601eb639597b8b0515a88cb5ac7fa8a4aabe3c87349d"]],"leaf": ["0x3e7","0x2d14dcb388a6678e5482d0a6ba25da4000a9196b2673f4312831877eb115486d","0x2d14dcb388a6678e5482d0a6ba25da4000a9196b2673f4312831877eb115486d","0x2d14dcb388a6678e5482d0a6ba25da4000a9196b2673f4312831877eb115486d"]}
+1;{"root":["0x29867d38d62eee29e0c4ad1bcb25826696348752d9644b6e056fd5cbd913c3d6"],"inPathIndices":[0],"inPathElements":[["0x0"]],"leaf":["0x20e54c83532f8c0c0d1c36e164435eff29445efc516b78a7ba57d73083d9bbe5"]}
```

Comment:
since you regenerated the test data can you explain again which test tests which property?

Reply:
Sure. 

1. Inclusion
1.1. Correct circuit:
https://github.com/Lightprotocol/light-protocol/blob/sergey/gnark-circuits-combined/gnark-prover/test-data/inclusion.csv#L1
1.2. Wrong root:
https://github.com/Lightprotocol/light-protocol/blob/sergey/gnark-circuits-combined/gnark-prover/test-data/inclusion.csv#L2
1.3. Wrong leaf:
https://github.com/Lightprotocol/light-protocol/blob/sergey/gnark-circuits-combined/gnark-prover/test-data/inclusion.csv#L3
1.4. Wrong inPathIndices: 
https://github.com/Lightprotocol/light-protocol/blob/sergey/gnark-circuits-combined/gnark-prover/test-data/inclusion.csv#L4
1.5. Wrong inPathElements: 
https://github.com/Lightprotocol/light-protocol/blob/sergey/gnark-circuits-combined/gnark-prover/test-data/inclusion.csv#L5

2. Non-inclusion.
2.1. Correct circuit:
https://github.com/Lightprotocol/light-protocol/blob/sergey/gnark-circuits-combined/gnark-prover/test-data/non_inclusion.csv#L1
2.2. Wrong root:
https://github.com/Lightprotocol/light-protocol/blob/sergey/gnark-circuits-combined/gnark-prover/test-data/non_inclusion.csv#L2
2.3. Wrong (low) value:
https://github.com/Lightprotocol/light-protocol/blob/sergey/gnark-circuits-combined/gnark-prover/test-data/non_inclusion.csv#L3
2.3. Wrong (high) value:
https://github.com/Lightprotocol/light-protocol/blob/sergey/gnark-circuits-combined/gnark-prover/test-data/non_inclusion.csv#L4
2.4. Wrong inPathIndices: 
https://github.com/Lightprotocol/light-protocol/blob/sergey/gnark-circuits-combined/gnark-prover/test-data/non_inclusion.csv#L5
2.5. Wrong inPathElements: 
https://github.com/Lightprotocol/light-protocol/blob/sergey/gnark-circuits-combined/gnark-prover/test-data/non_inclusion.csv#L6

3. Combined.
3.1. Correct circuit:
https://github.com/Lightprotocol/light-protocol/blob/sergey/gnark-circuits-combined/gnark-prover/test-data/combined.csv#L1
3.2. Wrong non-inclusion root:
https://github.com/Lightprotocol/light-protocol/blob/sergey/gnark-circuits-combined/gnark-prover/test-data/combined.csv#L2
3.3. Wrong non-inclusion (low) value:
[https://github.com/Lightprotocol/light-protocol/blob/sergey/gnark-circuits-combined/gnark-prover/test-data/combined.csv#L3](https://github.com/Lightprotocol/light-protocol/blob/sergey/gnark-circuits-combined/gnark-prover/test-data/non_inclusion.csv#L3)
3.3. Wrong non-inclusion (high) value:
[https://github.com/Lightprotocol/light-protocol/blob/sergey/gnark-circuits-combined/gnark-prover/test-data/combined.csv#L](https://github.com/Lightprotocol/light-protocol/blob/sergey/gnark-circuits-combined/gnark-prover/test-data/non_inclusion.csv#L3)4
3.4. Wrong non-inclusion inPathIndices: 
[https://github.com/Lightprotocol/light-protocol/blob/sergey/gnark-circuits-combined/gnark-prover/test-data/combined.csv#L](https://github.com/Lightprotocol/light-protocol/blob/sergey/gnark-circuits-combined/gnark-prover/test-data/non_inclusion.csv#L3)5
3.5. Wrong non-inclusion inPathElements: 
https://github.com/Lightprotocol/light-protocol/blob/sergey/gnark-circuits-combined/gnark-prover/test-data/combined.csv#L6
3.6. Wrong inclusion root:
https://github.com/Lightprotocol/light-protocol/blob/sergey/gnark-circuits-combined/gnark-prover/test-data/combined.csv#L7
3.7. Wrong leaf:
https://github.com/Lightprotocol/light-protocol/blob/sergey/gnark-circuits-combined/gnark-prover/test-data/combined.csv#L13
3.8. Wrong inPathIndices: 
https://github.com/Lightprotocol/light-protocol/blob/sergey/gnark-circuits-combined/gnark-prover/test-data/combined.csv#L19
3.9. Wrong inPathElements: 
https://github.com/Lightprotocol/light-protocol/blob/sergey/gnark-circuits-combined/gnark-prover/test-data/combined.csv#L25

Data generation happens [here](https://github.com/Lightprotocol/light-protocol/blob/sergey/gnark-circuits-combined/gnark-prover/merkle-tree/tree_test.go#L145).

Reply:
thanks!